modules/porous_flow/doc/content/source/materials/PorousFlowSingleComponentFluid.md

```diff
@@ -2,9 +2,149 @@
 
 !syntax description /Materials/PorousFlowSingleComponentFluid
 
+## Choice of units
+
+Most of MOOSE is independent of the choice of units, but the units must be kept consistent throughout the input file.
+
+- For instance, time can be measured in seconds, nano-seconds, years, etc, providing this choice is used everywhere.
+
+- For instance, you must specify fluid bulk modulus and solid Young's modulus using the same units (for example, both in GPa).
+
+However, the [FluidProperties module](fluid_properties/index.md) assumes that the pressure units are Pascals, the time units are seconds, temperature is measured in Kelvin, etc.  Therefore, when using this module, users are ordinarily restricted to using these standard SI units.  Almost all PorousFlow input files use the FluidProperties module.
+
+`PorousFlowSingleComponentFluid` allows different units to be employed: it modifies the input and output of the FluidProperties module to reflect the choice of units.
+
+- For instance, suppose you choose to measure temperature in Celcius.  Then `PorousFlowSingleComponentFluid` will add 273.15 to your temperature before feeding it to the FluidProperties module.  No other alterations are needed.
+
+- For instance, suppose you choose to measure pressure in MPa.  Then `PorousFlowSingleComponentFluid` will multiply your pressure by $10^6$ before feeding it to the FluidProperties module.  Since the viscosity returned by the FluidProperties module has units Pa.s, it needs to be converted to MPa.s before using it in the remainder of PorousFlow, so the FluidProperties viscosity is multiplied by $10^{-6}$.
+
 !alert note
-The FluidProperties userobjects expect temperature in Kelvin. If the simulation uses temperature in
-Celsius, `temperature_units = celsius` must be used.
+If you choose non-default SI units, you should ensure that all the other parameters in your input file use these non-default units.  For instance, if you choose to measure pressure in MPa, then your solid-mechanical stresses, Young's moduli, strengths, etc, must also be specified in MPa.  That is, `PorousFlowSingleComponentFluid` only interfaces with the FluidProperties module: it doesn't also change your pressure boundary conditions, end time, gravity, etc: all these must be specified by you in the units you choose.
```

Comment:
Maybe make it stronger like "you must ensure that"

And perhaps a note that PF cannot check consistency of units automatically?